Virtual cockpit

Jordan-64 opened the pull request:
New Ordeal knob for CMVC and LMVC.
New frame for LM CW lights and new texture.
New EMS Lift Vector indicators.
Add LM VC Tapemeter Lighting.
All single lights CW, DSKY etc are now independent lit.
New FloodLighting in CM and LM VC
Added animations for the Ordeal, Altimeter Pluger, Wastedisposal and Panel382 Cover.
New Round Meters and Needles in CM. eg Cryo Tank, SPS Propelant Fuell Cell  etc.
More Accurate Font for the DSKY and DEDA.
Added Optics Sextant and Telescope mesh in CMVC with stowing animation.

Jordan-64 committed:
CMVC New Texture for the Ordeal and new Ordeal Knob
LMVC DSKY and CMVC CW Lighting Update.

Jordan-64 committed:
CMVC New Texture for the Ordeal and new Ordeal Knob
LMVC DSKY and CMVC CW Lighting Update.

Jordan-64 committed:
All lighting code merged into one place but separate for the CM and LM.
#ifdef _OPENORBITER reduced to the bare minimum.
LM CW lights initialized, working with some limitations.

Jordan-64 committed:
CMVC Lights (Master Alarm, LV Engines, Abort and LiftOff No Auto Abort)  now Lit at full brightness and are not Dimmable.

indy91 committed:
Merge branch 'Orbiter2016' into VirtualCockpit

# Conflicts:
#	Meshes/ProjectApollo/CM-VC.msh
#	Meshes/ProjectApollo/LM_VC.msh
#	Orbitersdk/samples/ProjectApollo/src_csm/saturnvc.cpp

Jordan-64 committed:
Merge pull request #8 from indy91/VirtualCockpit

Fix merge conflicts from latest Orbiter2016 branch updates

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Lighting Update for CM and LM

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Fix LMVC_emis.dds beeing to large

Jordan-64 committed:
CMVC Fix FDAI ball beeing to dark
LMVC make AGS OPR ERR full lit

Jordan-64 committed:
New EMS Lift Vector indicators

Jordan-64 committed:
Replaced the old NASSP LEB Mission Timer frame with new new one + Connected Panel 306 to Panel 100 Integral rheostat.

Jordan-64 committed:
Tilt CMVC attitude numbers by 45°

Jordan-64 committed:
FIX CMVC Caution and Warning lights nor working properly

Jordan-64 committed:
FIX Overflow bug, CW Lights LV Engines parts.

Jordan-64 committed:
CMVC Lighting Updates.
SPS Thrust, .05G, Lift Vector Indicators and Masteralarm should now work.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
VC Lighting Updates + New Texture for LM Caution & Warning Lights

rcflyinghokie committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP.git into VirtualCockpit

Jordan-64 committed:
Add LM VC Tapemeter Lighting

Jordan-64 committed:
Add LM VC RCS Helium Press x10 light. Big thanks for Coding help to @rcflyinghokie

Jordan-64 committed:
Fix LMVC Ordeal

Jordan-64 committed:
SetVCLighting Function Overloading for changing Meshes with only one Material.
Change the Tapemeter material so that it remains “light gray” when turned off.

Jordan-64 committed:
Added new Caution & Warning Lights Frame in LM
Added Lighting for the LEB Conditional Lamps

Jordan-64 committed:
Made Lighting for the LEB Conditional Lamps dimmable.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Added CMVC LVRateLight, LVGuidLight and SIISepState Lights
Change the Texture for the Caution Warning in LMVC

Jordan-64 committed:
Fix CMVC Caution&Warning Lights not work correct when CSM/CM switch is in position CM

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Added a few missing screws in CMVC
Added check for empty materials in CMVC Caution & Warning lists.
Made the black frame around the Caution & Warning lights in LMVC a bit smaller.
Initial texturing for Panel 122 in CMVC

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Reworking on CMVC Panel 122
Add Switch guards in CMVC Panel 225,226 and 229 to avoid the ugly texture

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Changed DSKY & DEDA font to more accurate one.
Additional Modeling & Texturing in Panel 122 in CMVC.
Modeled Waste Disposal Knob in CMVC which was only textured.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Disconnect CMVC DSKY Status lights from integral Rheostat and connect them to Nummerics Rheostat

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
CMVC Lighting Code Changes.
Connected Panel 225 and 226 to Panel 100 integral rheostat.
Made Panel 122 Condition Lamps fixed brightness.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
New more accurate Texture, with removed "400" text, for CM Cryogenic Tank meters.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

rcflyinghokie committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP.git into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'Orbiter2016' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Added Altimeter Cover
WIP Testing new Gauge meters

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
*** IMPORTANT *** METERS ARE STILL WIP(TEXTURES ARE NOT FINISHED. USE THIS BRANCH FOR NOW ONLY FOR TESTING.
CMVC Ordeal, Altimeter cover, and Waste cover can now be removed.

Jordan-64 committed:
Added Animation for CMVC Waste disposal cover.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Added animated Cover in CMVC Coolant Control Panel 382

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Animations Code for the Waste Disposal lid and Panel 382 cover changed.

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Added animation for the LM Antenna Handle.

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

rcflyinghokie commented:
Is there any reason this still needs to be draft?

Jordan-64 commented:
> Is there any reason this still needs to be draft?

I wanted to finish the cm meters on panel 2 and 3 first. 
Actually, I'm waiting until someone wants to merge it. Is it stable enough? If so, then I can finish the meters later.

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Removed Code for  the EVA Antenna Handle to avoid merge Conflicts with the LM_Additions.
Added animation for the Altimeter Cover in CM.

Jordan-64 committed:
Maybe fix for merge Conflicts, I don't know. Just testing

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
CMVC CHANGES
+Adding new CM VC Lighting
+Flood Light Knobs (Panel 5,8 and LEB) now Controls the Lighting
+Improved Switch Guard and holders
+Better Gauges *** Needs Code Changes

Jordan-64 committed:
Disable Debug String in CM VC Lighting

Jordan-64 committed:
Added stowed animation for the CM Ordeal

rcflyinghokie committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP.git into VirtualCockpit

rcflyinghokie committed:
Scale Temp/Press meter row

rcflyinghokie committed:
Add cryo row meter scaling

rcflyinghokie committed:
remove duplicate draw line

rcflyinghokie committed:
Ad VC overload for O2 cryo

rcflyinghokie committed:
Adjust scaling per systems handbook

rcflyinghokie committed:
revert decimal removal to reduce change

rcflyinghokie committed:
add debug lines, comment out

rcflyinghokie committed:
Add textures

rcflyinghokie committed:
Add fixed scaling to meters

rcflyinghokie committed:
comment out debug lines

rcflyinghokie committed:
fix display value call

rcflyinghokie committed:
Revert textures to VC branch

rcflyinghokie committed:
Sync rotation angle in VC displays

Jordan-64 committed:
Updated Textures for the new CMVC Gauge Meters

rcflyinghokie committed:
Add voltage transducers directly to fuel cell

rcflyinghokie committed:
Add scaling for FC and SPS temp meters

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Fix CMVC Ordeal animation. Mesh overlapped with couch armrest.
Add LMVC Overhead floodlights.

rcflyinghokie committed:
Update meter scaling for fuel cell flows

rcflyinghokie committed:
update visual files to match repo

Jordan-64 committed:
Merge pull request #9 from rcflyinghokie/VCDisplayScaling

Add display scaling for 2D and VC Cryo meter row and Cabin/Suit ECS meter row

Jordan-64 committed:
Fix Angles and position for the Needles in CM RCS meters and SPS PRPLNT TANK + FUELL CELL meters

rcflyinghokie committed:
Correct bad math on VC draw for cabin/suit press meter

Jordan-64 committed:
Merge pull request #10 from rcflyinghokie/VCDisplayScaling

Correct bad math on VC draw for cabin/suit press meter

Jordan-64 committed:
Fix LM VC Hose hides some Rheostats in ECS
Fix LM VC DSKY Lights and CW Lights Lit even when unpowered

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

rcflyinghokie reviewed (dismissed):
Testing this works very well, mathematical models provide the best fit for the gauges right now until we have a better ides of how they actually handled the non linear scaling.

TINYBOB1 reviewed (dismissed):
Works as intended.  animations all check out, nothing unexpected

Jordan-64 committed:
CMVC Animations Code changed to make it easier to read. Old Code is still there but commented out.
Change Color bands in CMVC Round meters to more accurate. Both for the VC and 2D
Added Mesh and animations for the Glareshades in CMVC

Jordan-64 committed:
Upload missing textures

rcflyinghokie reviewed (changes requested):
bmp files for 2D CM cockpit will not load on this revision

TINYBOB1 commented:
In what way does it not load?  I'm not available to test right now, just curious what the symptoms are.

Jordan-64 committed:
Fix. Bitmaps in CM are not loaded due to wrong compression.
Fix. Glareshade status in CM VC is not saved.

Jordan-64 committed:
Fix some self illuminating needles in CM and LM VC
Add Velcros for DV EMS Glare shade
More accurate Glare shades for CMVC
Fix hose hiding some of the ECS Rheostats in LMVC

Jordan-64 committed:
Added animated Optics(Sextant) in CM VC
Changed the color for the Comp Lights in LM VC to match better the original. Thanks to @CaptainSwag101 for the RGB Colors.

Jordan-64 commented:
> bmp files for 2D CM cockpit will not load on this revision

Jordan-64 reviewed (commented):
bmp's not loading fixed.

rcflyinghokie commented:
> In what way does it not load? I'm not available to test right now, just curious what the symptoms are.

compression

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

Jordan-64 committed:
Added Optics Telescope mesh with animation in CM VC

Jordan-64 committed:
Corrected color bands in CM 2D BMPs that were grayish instead of green

jalexb88 commented:
Once I land my Apollo 12 LM on the moon I will make my review...so far so good up to LM activation.

jalexb88 reviewed (dismissed):
Tested from launch to lunar landing on Apollo 12 with O2024. No major issues.

rcflyinghokie commented:
I will check the other gauges this week and then relist my approval

Jordan-64 committed:
Fix Merge Conflict. Test.

Jordan-64 committed:
Merge https://github.com/Jordan-64/NASSP into VirtualCockpit

# Conflicts:
#	Orbitersdk/samples/ProjectApollo/src_lm/lemvc.cpp

Jordan-64 committed:
FIX Merge Conflict with LM Additions

Jordan-64 committed:
Fixed missing vertex creating gaps in CMVC LEB.

Jordan-64 committed:
FIX incorrect LMVC ECS mesh normals and UVs causing transparency so the back side of the ECS was visible.

rcflyinghokie committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP.git into VirtualCockpit

rcflyinghokie committed:
Change component light "off" tip color to be slightly darker than rest of cylinder

Jordan-64 committed:
Merge pull request #11 from rcflyinghokie/VirtualCockpit

Change component light "off" tip color to be slightly darker than rest of cylinder

Jordan-64 committed:
FIX incorrect UVs in LMVC Panel1 causing texture distortion.
Made LMVC Comp Lights socket longer.

Jordan-64 committed:
FIX Merge Conflict

Jordan-64 committed:
Replace spaces with TAB's

CaptainSwag101 commented:
I will try to fly my next mission using this branch, and see how everything works. I might see about creating a custom font for the cockpit EL displays like the LM numeric displays and the CM mission timers, since we now know those were likely both a different color and used a different font.

Jordan-64 committed:
Added Side Hatch View in VC. Can be switched to with CTRL+ Left, Right, Up and Down from Window 2 and 4, also from Panel 2 Center View.

Jordan-64 committed:
Fix Gap in CMVC Hull behind the Ordeal, reported by @jalexb88
Fix missing Switch base in CMVC Panel3, reported by @TINYBOB1

Jordan-64 committed:
Merge branch 'orbiternassp:Orbiter2016' into VirtualCockpit

CaptainSwag101 committed:
Replace digit textures for non-DSKY and non-EMS displays
This does not yet adjust the relevant code which draws these displays, which will be broken temporarily due to the digit sizes changing.

CaptainSwag101 committed:
Edit 2D panel bitmaps, improve EL background color and size

CaptainSwag101 committed:
Small CM 2D panel adjustments, label tweaks

CaptainSwag101 committed:
CM 2D panel area & mission timer code changes

CaptainSwag101 committed:
LM panel area and code changes

CaptainSwag101 committed:
Disable interpolation for EMS digital display

CaptainSwag101 committed:
Add external timing power to the mission timer
This will determine whether or not the "tuning fork" should be displayed, if that external timing fails.

Jordan-64 committed:
Resize the Accelerometer and Altimeter cover click points to match the coverage size.

Jordan-64 committed:
Merge branch 'VirtualCockpit' of https://github.com/Jordan-64/NASSP into VirtualCockpit

CaptainSwag101 committed:
Implement Mission Timer external timing source

CaptainSwag101 committed:
Adjust scaling of some numeric displays, resize LM oxid/fuel gauge in 2D

CaptainSwag101 committed:
Update background color for numeric gauges

CaptainSwag101 committed:
Improve spacing of LM event timer digits

CaptainSwag101 committed:
Extend LM timer label lines

Jordan-64 committed:
Fix CMVC GFXbug on the S1B engine lights, reported by @maxq519

rcflyinghokie committed:
Cond Exh and Skin Temp Return 0V when SM is jettisoned

rcflyinghokie committed:
Add safety comment

Jordan-64 committed:
Merge pull request #12 from rcflyinghokie/VCDisplayScaling

Cond Exh and Skin Temp Return 0V when SM is jettisoned

CaptainSwag101 committed:
Add latest VC textures/meshes from Jordan

CaptainSwag101 committed:
Update VC digit positions to match new texture work

CaptainSwag101 committed:
Update digit background color blending

CaptainSwag101 committed:
Move 2D CSM LEB mission timer digits down 1px

Jordan-64 committed:
LMVC Preparation for @CaptainSwag101 new Mission timer Digits

Jordan-64 committed:
CMVC Preparation for @CaptainSwag101 new Mission timer Digits

CaptainSwag101 committed:
Merge in Jordan's latest changes

CaptainSwag101 committed:
Move CM LEB MT digits down by 1px

Jordan-64 committed:
Merge pull request #13 from CaptainSwag101/VirtualCockpit-NewDigits

Add new EL digits

CaptainSwag101 reviewed (dismissed):
Everything looks fantastic!

maxq519 reviewed (dismissed):
Everything looks amazing!

ThatRedMelon reviewed (dismissed):
Looks gorgeous!

CaptainSwag101 committed:
Misc code fixes
Fix the tuning fork logic for CM panel 306, and fix the double-to-int build warnings

indy91 commented on Orbitersdk/samples/ProjectApollo/src_sys/missiontimer.cpp:
As discussed on Discord, this file needs the build warnings fixed and the tuning fork code added.

CaptainSwag101 replied:
PR outstanding to fix these issues.

Jordan-64 committed:
Merge pull request #14 from CaptainSwag101/VirtualCockpit-NewDigits

Misc code fixes

TINYBOB1 reviewed (dismissed):
Some well needed and amazing looking upgrades!

Jordan-64 committed:
Bug fix. The VC FDAI roll arrow overlaps the increment marking. #1351

Jordan-64 committed:
2 screws in the CMVC engines that were too many were removed

indy91 committed:
Improve handling of new CM VC crew equipment (covers, glareshades etc:); Remove redundant and unused code; Initialize more variables; Fix VC bugs

Jordan-64 committed:
Merge pull request #15 from indy91/VirtualCockpit

Improve handling of new CM VC crew equipment (covers, glareshades etc…

indy91 committed:
Delete more unused variables and move others out of the Saturn/LEM classes

Jordan-64 committed:
Merge pull request #16 from indy91/VirtualCockpit

Delete more unused variables and move others out of the Saturn/LEM cl…

maxq519 reviewed (approved):
All looks good

ThatRedMelon reviewed (approved):
Hopefully the final time I approve this, all still looks great! 😆
